Assets/Scripts/PlayerCharacter.cs

```diff
@@ -0,0 +1,211 @@
+        // Debug.Log("IsOnGround?: " + isOnGround);
+
+    }
+    private void UpdateHorizontalInput()
+    {
+        horizontalInput = Input.GetAxisRaw("Horizontal");
+    }
+    private void HandleJumpInput()
+    {
+        if (Input.GetButtonDown("Jump") & isOnGround)
+        {
+            rb2d.AddForce(Vector2.up * jumpForce, ForceMode2D.Impulse);
+        }
+    }
+    private void Move()
+    {
+        rb2d.AddForce(Vector2.right * horizontalInput * accelerationforce);
+        Vector2 clampedVelocity = rb2d.velocity;
+        clampedVelocity.x = Mathf.Clamp(rb2d.velocity.x, -maxSpeed, maxSpeed);
+        rb2d.velocity = clampedVelocity;
+    }
+    public void Flip()
+    {
+        facingRight = !facingRight;
+        Vector3 theScale = transform.localScale;
+        theScale.x *= -1;
+        transform.localScale = theScale;
+    }
+    public void AreTheyDead()
+    {
+        switch (isDead)
+        {
+            case false:
+
+                break;
+
+            case true:
+
+                
+                rb2d.constraints = RigidbodyConstraints2D.FreezeAll;
+                anim.SetBool("isDead", true);
+                deathText.text = "Press R to respawn";
+
+                if (Input.GetKeyUp(KeyCode.R))
+                {
+                    isDead = false;
+                    Debug.Log("R key pressed");
+                    Unfreeze();
+                    Respawn();
+                }
+                break;
+        }
+    }
+    public void Respawn()
+    {
+
+        deathText.text = " ";
+            if (currentCheckpoint == null)
+                SceneManager.LoadScene(SceneManager.GetActiveScene().name);
+            else
+            {
+                rb2d.velocity = Vector2.zero;
+                transform.position = currentCheckpoint.transform.position;
+            }
+    }
+    public void Unfreeze()
+    {
+        anim.SetBool("isDead", false);
+        rb2d.constraints = RigidbodyConstraints2D.None;
+        rb2d.constraints = RigidbodyConstraints2D.FreezeRotation;
+    }
+    private void SetScoreCount()
+    {
+        countText.text = "Count: " + scoreCount.ToString();
+        deathText.text = " ";
+    }
+    public void SetCurrentCheckpoint(Checkpoint newCurrentCheckpoint)
+    {
+        if(currentCheckpoint != null)
+            currentCheckpoint.SetIsActivated(false);
+
+        currentCheckpoint = newCurrentCheckpoint;
+        currentCheckpoint.SetIsActivated(true);
+    }
+    void OnTriggerEnter2D(Collider2D other)
+    {
+        if (other.gameObject.CompareTag("Collectable"))
+        {
+            collectableSound.Play();
+            other.gameObject.SetActive(false);
+            scoreCount++;
+            SetScoreCount();
+        }
+
+        if (other.gameObject.CompareTag("Hazard"))
+        {
+            isDead = true;
+        }
+
+        if (other.gameObject.CompareTag("Ring"))
+        {
+            ringSound.Play();
+            other.gameObject.SetActive(false);
```

Comment:
Looks good.

Minor thing: there’s a lot of extra lines that aren’t being used. Maybe you should delete those lines and spaces.